pkg/hubble/observer/local_observer.go

```diff
@@ -197,7 +197,7 @@ func (s *LocalObserverServer) GetStopped() chan struct{} {
 }
 
 // GetPayloadParser implements GRPCServer.GetPayloadParser.
-func (s *LocalObserverServer) GetPayloadParser() *parser.Parser {
+func (s *LocalObserverServer) GetPayloadParser() parser.Decoder {
```

Comment:
@rolinh IIUC `LocalObserver` doesn't know the actual type though, since after this patch it is constructed with the interface given to `NewLocalServer`.

Reply:
Yeah, now the LocalObserver constructor accepts the interface to construct the object. It doesn't accept a struct any longer. Similar to NamespaceManager.

Also, the GetPayloadParser is not a member function of the parser class. The New method does return the struct (L41 in parser.go).